CMakeLists.txt

```diff
@@ -469,6 +487,16 @@ macro(nebula_add_library name type)
     )
 endmacro()
 
+# Generate object or shared lib
+macro(nebula_add_all_library name)
+    if (NOT ENABLE_ONLY_SHARED)
```

Comment:
The logic is right?
If ENABLE_TESTING is true and ENABLE_ONLY_SHARED is false, what happens? link them both?